parser/prism/Translator.cc

```diff
@@ -1160,19 +1160,31 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node, bool preserveCon
             ENFORCE(arg != nullptr);
             ENFORCE(arg->hasDesugaredExpr());
 
+            auto valueNode = definedNode->value;
+
             // Desugar to `::Magic.defined_instance_var(ivar)` or `::Magic.defined_class_var(cvar)`
-            if (auto *ivar = parser::NodeWithExpr::cast_node<parser::IVar>(arg.get())) {
-                auto sym = MK::Symbol(arg->loc, ivar->name);
-                auto expr = MK::Send1(arg->loc, MK::Magic(arg->loc), core::Names::definedInstanceVar(),
+            if (PM_NODE_TYPE_P(valueNode, PM_INSTANCE_VARIABLE_READ_NODE)) {
```

Comment:
I wonder if it's possible to give the same treatment to the `constPathNode` below?

If so, we can replace this if ladder with a classic `switch (PM_NODE_TYPE(valueNode))`